File: 219.contains-duplicate-ii.go
func min(a int, b int) int {
	if a < b {
		return a
	}

	return b
}

func abs(a int) int {
	if a >= 0 {
		return a
	}
	return -a
}

func containsNearbyDuplicate(nums []int, k int) bool {
	res := len(nums)

	for i := 0; i < len(nums); i++ {
		for j := i+1; j < len(nums); j++ {
			if nums[i] == nums[j] {
				res = min(abs(j-i), res)
			}
		}
	}

	return res <= k && len(nums) >= 2 && res < len(nums)
}

File: 219.contains-duplicate-ii.cpp
// CreateTime: 2019-12-02 10:43:51
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> m;

        int len = nums.size();
        for (int i = 0; i < len; i++) {
            if (m.find(nums[i]) == m.end()) {
                m[nums[i]] = i;
            } else {
                if (i - m[nums[i]] <= k) {
                    return true;
                }
                if (i - m[nums[i]] > k) {
                    m[nums[i]] = i;
                }
            }
        }
        
        return false;
    }
};
